src/mon/MDSMonitor.cc

```diff
@@ -668,6 +670,13 @@ bool MDSMonitor::prepare_beacon(MonOpRequestRef op)
           pending_daemon_health_rm.insert(erased_gid);
         }
       }
+
+      auto fscid = pending_fsmap.mds_roles.at(gid);
```

Comment:
This results in an assert because that gid has been removed from the map (a few lines earlier):

http://pulpito.front.sepia.ceph.com/pdonnell-2017-07-26_18:40:14-fs-wip-pdonnell-testing-20170725-distro-basic-smithi/1448145/

````
2017-07-27T07:45:54.794 INFO:tasks.ceph.mon.a.smithi200.stderr:terminate called after throwing an instance of 'std::out_of_range'
2017-07-27T07:45:54.857 INFO:tasks.ceph.mon.a.smithi200.stderr:  what():  map::at
2017-07-27T07:45:54.857 INFO:tasks.ceph.mon.a.smithi200.stderr:*** Caught signal (Aborted) **
2017-07-27T07:45:54.857 INFO:tasks.ceph.mon.a.smithi200.stderr: in thread 7f6f15d0a700 thread_name:ms_dispatch
2017-07-27T07:45:54.931 INFO:tasks.ceph.mon.a.smithi200.stderr: ceph version 12.1.1-575-g7776c35 (7776c35bf2240350c9710c578bb27c65928ccee1) luminous (rc)
2017-07-27T07:45:54.931 INFO:tasks.ceph.mon.a.smithi200.stderr: 1: (()+0x936fd4) [0x55d305ac8fd4]
2017-07-27T07:45:54.931 INFO:tasks.ceph.mon.a.smithi200.stderr: 2: (()+0x11390) [0x7f6f1d831390]
2017-07-27T07:45:54.931 INFO:tasks.ceph.mon.a.smithi200.stderr: 3: (gsignal()+0x38) [0x7f6f1bf81428]
2017-07-27T07:45:54.931 INFO:tasks.ceph.mon.a.smithi200.stderr: 4: (abort()+0x16a) [0x7f6f1bf8302a]
2017-07-27T07:45:54.932 INFO:tasks.ceph.mon.a.smithi200.stderr: 5: (__gnu_cxx::__verbose_terminate_handler()+0x16d) [0x7f6f1c8c384d]
2017-07-27T07:45:54.932 INFO:tasks.ceph.mon.a.smithi200.stderr: 6: (()+0x8d6b6) [0x7f6f1c8c16b6]
2017-07-27T07:45:54.932 INFO:tasks.ceph.mon.a.smithi200.stderr: 7: (()+0x8d701) [0x7f6f1c8c1701]
2017-07-27T07:45:54.932 INFO:tasks.ceph.mon.a.smithi200.stderr: 8: (()+0x8d919) [0x7f6f1c8c1919]
2017-07-27T07:45:54.932 INFO:tasks.ceph.mon.a.smithi200.stderr: 9: (std::__throw_out_of_range(char const*)+0x3f) [0x7f6f1c8ea2cf]
2017-07-27T07:45:54.932 INFO:tasks.ceph.mon.a.smithi200.stderr: 10: (std::map<mds_gid_t, int, std::less<mds_gid_t>, std::allocator<std::pair<mds_gid_t const, int> > >::at(mds_gid_t const&)+0x52) [0x55d3057656f2]
2017-07-27T07:45:54.933 INFO:tasks.ceph.mon.a.smithi200.stderr: 11: (MDSMonitor::prepare_beacon(boost::intrusive_ptr<MonOpRequest>)+0x2119) [0x55d305755849]
2017-07-27T07:45:54.933 INFO:tasks.ceph.mon.a.smithi200.stderr: 12: (MDSMonitor::prepare_update(boost::intrusive_ptr<MonOpRequest>)+0x20c) [0x55d30575ea7c]
2017-07-27T07:45:54.933 INFO:tasks.ceph.mon.a.smithi200.stderr: 13: (PaxosService::dispatch(boost::intrusive_ptr<MonOpRequest>)+0x9b4) [0x55d3056afa24]
2017-07-27T07:45:54.933 INFO:tasks.ceph.mon.a.smithi200.stderr: 14: (Monitor::dispatch_op(boost::intrusive_ptr<MonOpRequest>)+0x76d) [0x55d305580bbd]
2017-07-27T07:45:54.934 INFO:tasks.ceph.mon.a.smithi200.stderr: 15: (Monitor::_ms_dispatch(Message*)+0x6db) [0x55d305581eab]
2017-07-27T07:45:54.993 INFO:tasks.ceph.mon.a.smithi200.stderr: 16: (Monitor::ms_dispatch(Message*)+0x23) [0x55d3055af383]
2017-07-27T07:45:54.993 INFO:tasks.ceph.mon.a.smithi200.stderr: 17: (DispatchQueue::entry()+0xf4a) [0x55d305a70a1a]
2017-07-27T07:45:54.993 INFO:tasks.ceph.mon.a.smithi200.stderr: 18: (DispatchQueue::DispatchThread::entry()+0xd) [0x55d30587df7d]
2017-07-27T07:45:54.993 INFO:tasks.ceph.mon.a.smithi200.stderr: 19: (()+0x76ba) [0x7f6f1d8276ba]
2017-07-27T07:45:54.993 INFO:tasks.ceph.mon.a.smithi200.stderr: 20: (clone()+0x6d) [0x7f6f1c05282d]
2017-07-27T07:45:54.994 INFO:tasks.ceph.mon.a.smithi200.stderr:2017-07-27 07:45:54.806696 7f6f15d0a700 -1 *** Caught signal (Aborted) **
````